vital/optional.h

```diff
@@ -9,991 +9,1275 @@
+  {
+  }
 
-  ~storage_t(){}
+  ~storage_t()
+  {
+  }
 };
 
-
-template <class T>
+template < class T >
 union constexpr_storage_t
 {
-    unsigned char dummy_;
-    T value_;
+  unsigned char dummy_;
+  T value_;
 
-    constexpr constexpr_storage_t( trivial_init_t ) noexcept : dummy_() {};
+  constexpr constexpr_storage_t( trivial_init_t ) noexcept : dummy_()
+  {
+  };
 
-    template <class... Args>
-    constexpr constexpr_storage_t( Args&&... args ) : value_(constexpr_forward<Args>(args)...) {}
+  template < class... Args > constexpr constexpr_storage_t( Args &&
+                                                            ... args ) : value_( constexpr_forward< Args >(
+                                                                                   args )... )
+  {
+  }
 
-    ~constexpr_storage_t() = default;
+  ~constexpr_storage_t() = default;
 };
 
-
-template <class T>
+template < class T >
 struct optional_base
 {
-    bool init_;
-    storage_t<T> storage_;
+  bool init_;
+  storage_t< T > storage_;
 
-    constexpr optional_base() noexcept : init_(false), storage_(trivial_init) {};
+  constexpr optional_base() noexcept : init_( false ), storage_( trivial_init )
+  {
+  }
 
-    explicit constexpr optional_base(const T& v) : init_(true), storage_(v) {}
+  explicit constexpr optional_base( const T& v ) : init_( true ), storage_( v )
+  {
+  }
 
-    explicit constexpr optional_base(T&& v) : init_(true), storage_(constexpr_move(v)) {}
+  explicit constexpr optional_base( T&& v ) : init_( true ), storage_( constexpr_move(
+                                                                         v ) )
+  {
+  }
 
-    ~optional_base() { if (init_) storage_.value_.T::~T(); }
+  ~optional_base() { if( init_ ) { storage_.value_.T::~T(); } }
 };
 
-
-template <class T>
+template < class T >
 struct constexpr_optional_base
 {
-    bool init_;
-    constexpr_storage_t<T> storage_;
+  bool init_;
+  constexpr_storage_t< T > storage_;
 
-    constexpr constexpr_optional_base() noexcept : init_(false), storage_(trivial_init) {};
+  constexpr constexpr_optional_base() noexcept : init_( false ), storage_(
+                                                   trivial_init ) {}
 
-    explicit constexpr constexpr_optional_base(const T& v) : init_(true), storage_(v) {}
+  explicit constexpr constexpr_optional_base( const T& v ) : init_( true ),
+                                                             storage_( v ) {}
 
-    explicit constexpr constexpr_optional_base(T&& v) : init_(true), storage_(constexpr_move(v)) {}
+  explicit constexpr constexpr_optional_base( T&& v ) : init_( true ),
+                                                        storage_( constexpr_move(
+                                                                    v ) ) {}
 
-    ~constexpr_optional_base() = default;
+  ~constexpr_optional_base() = default;
 };
 
-template <class T>
+template < class T >
```

Comment:
Bah, I give up on this file; the whole thing should be going away (C++17) anyway...